src/ownArrayList.java

```diff
@@ -0,0 +1,189 @@
+import java.util.Arrays;
+
+public class ownArrayList<T> implements List {
+
+    private final int INITIALIZE_CAPACITY = 3;
+
+    private int capacity;
+    private int newCapacity;
+    private int oldCapacity;
+    private int lastIndex;
+    private T[] valuesArray;
+
+
+    public ownArrayList() {
+        valuesArray = (T[]) new Object[INITIALIZE_CAPACITY];
+        this.capacity = INITIALIZE_CAPACITY;
+        newCapacity = INITIALIZE_CAPACITY;
+        lastIndex = 0;
+    }
+
+    public ownArrayList(int capacity) {
+        valuesArray = (T[]) new Object[capacity];
+        this.capacity = capacity;
+        newCapacity = capacity;
+        lastIndex = 0;
+
+    }
+
+    @Override
+    public void add(Object value) {
+        if (valuesArray[capacity - 1] != null) {
```

Comment:
optional: you can write
if(valuesArray.lenght <= lastIndex)

Reply:
this is check for empty cell in the array 
I don't understand, what will do :"if(valuesArray.lenght <= lastIndex)"